test/util.js

```diff
@@ -513,3 +515,261 @@ function checkCursorPaginatedHistoryObject ({
+ * @param {Object[]} [params.filters[i].customTestValues] - if specified, will only use those values
+ *   and test each one of those values
+ *   otherwise, will perform a fetch of objects and test with fixtures values
+ * @param {Boolean}  [params.filters[i].noResultsExistenceCheck = false]
+ *   by default, will check if there are results while it should have (based on auto-fetched values)
+ */
+async function checkFilters ({
+  t,
+  authorizationHeaders,
+  endpointUrl,
+  fetchEndpointUrl,
+  checkPaginationObject,
+  filters,
+}) {
+  const getEndpoint = (endpointUrl, queryParams) => {
+    if (endpointUrl.includes('?')) return `${endpointUrl}&${queryParams}`
+    else return `${endpointUrl}?${queryParams}`
+  }
+
+  const getRandomValue = (values) => _.first(_.shuffle(values))
+
+  const sortValues = (values) => {
+    return values.slice(0).sort((a, b) => {
+      if (a < b) return -1
+      else if (a > b) return 1
+      else return 0
+    })
+  }
+
+  // prop=value
+  const getExactValueFilter = (prop, exactValue) => `${prop}=${exactValue}`
+
+  // prop[]=value1&prop[]=value2
+  const getArrayValuesFilter = (prop, values) => values.map(v => `${prop}[]=${v}`).join('&')
+
+  // prop=value1,value2
+  const getArrayCommaSeparatedValuesFilter = (prop, values) => `${prop}=${values.join(',')}`
+
+  // prop[gte]=min&prop[lte]=max
+  const getRangeFilter = (prop, range) => `${prop}[gte]=${range.gte}&${prop}[lte]=${range.lte}`
+
+  const addCheckObjFunctions = (filter) => {
+    const {
+      prop,
+      customGetValue,
+      isArrayFilter,
+      isRangeFilter,
+      customCheck
+    } = filter
+
+    const getValue = (obj) => _.isFunction(customGetValue) ? customGetValue(obj) : obj[prop]
+    filter.getValue = getValue
+
+    filter.getExactValueCheckObj = _.curry((exactValue, obj) => {
+      return _.isFunction(customCheck)
+        ? customCheck(obj, exactValue)
+        : getValue(obj) === exactValue
+    })
+
+    if (isArrayFilter) {
+      filter.getArrayValuesCheckObj = _.curry((values, obj) => {
+        return _.isFunction(customCheck)
+          ? customCheck(obj, values)
+          : values.includes(getValue(obj))
+      })
+    } else if (isRangeFilter) {
+      filter.getRangeValuesCheckObj = _.curry((rangeValues, obj) => {
+        return _.isFunction(customCheck)
+          ? customCheck(obj, rangeValues)
+          : rangeValues.gte <= getValue(obj) && getValue(obj) <= rangeValues.lte
+      })
+    }
+  }
+
+  const addTestValues = (filter) => {
+    const {
+      isArrayFilter,
+      isRangeFilter,
+      customTestValues,
+    } = filter
+
+    addCheckObjFunctions(filter)
+
+    const getValue = filter.getValue
+
+    const useCustomValues = customTestValues && customTestValues.length
+
+    const realValues = _.compact(_.uniq(results.map(getValue)))
+    let testValues = useCustomValues
+      ? customTestValues
+      : _.shuffle(realValues).slice(0, 2) // only take few values
```

Comment:
Why only 2? Takes so much time?

Reply:
I restrict to only two different values because some fixtures don't have more than 2 different values for the specified filter.
Without restricting the values enough, it's like requesting all values with the array filter and we cannot be sure if the filter works or not.

`values.includes(getValue(obj))` would be similar to `allValues.includes(getValue(obj))` and the condition will always be true.
https://github.com/stelace/stelace/blob/47a66145b591f0c5e35d0dd3ac5c5be1107d1ba4/test/util.js#L602-L606

Reply:
> I restrict to only two different values because some fixtures don't have more than 2 different values for the specified filter.

Reply:
Yes indeed. For instance, webhook logs statuses can only have two values `success` or `error`.
https://github.com/stelace/stelace/blob/dcea4e38bed6462058fa11e10f6df92ddd1d9463/src/services/webhook.js#L437-L443

Maybe I can add a parameter to `checkFilters` to specify the number of values to use when retrieving automatically from fixtures to handle that special case and that won't affect all array filters, so we can increase the default value from 2 to 5.

Reply:
It’s still not clear why the fixtures should not only include filter values handled by the API endpoint, such as `success` and `error` for webhook log `status`?

Reply:
I meant the webhook log fixtures will only include `success` and `error` as `status` values, those are handled by the API. However in this special case where there are very few different allowed values, checking with a high number of filter values will be equivalent of the absence of the filter in the test. That's why I only took 2 values by default.

But a better solution might to apply specify `customTestValues: ['error', 'success']` for this case so the `checkFilter` function will individually check those values one by one.

If we increase the number of filter values from 2 to 5, we must use `customTestValues` for each filter whose the number of different values allowed by API is inferior to 5.

Reply:
Thanks for the explanation. The problem with your random approach is that tests may or may not fail depending on tested filter values, when they are expected to fail due to some regression.

Explicit is better than implicit. But you may have good reasons to only pick some filter values, if there are hundreds of them and you don’t want to test them all because there is no reason to have a regression for a particular value.

This is clearly not the case for discrete values with specific behaviors to test such as wehook log `status`.